### PR TITLE
Add support for timezone and hardware clock

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
Add support for timedatectl to set timezone and hardware clock

#### This Pull Request (PR) fixes the following issues
Fixes: #377

### CODE REVIEW

**Comment by jcpunk** [comment]
I almost went with

```puppet
  if $set_local_rtc {
    exec { 'set local hardware clock to local time':
      command => 'timedatectl set-local-rtc 1',
      onlyif  => 'test $(timedatectl show | grep LocalRTC | cut -d= -f2) = no',
      path    => $facts['path'],
    }
  } else {
    exec { 'set local hardware clock to UTC time':
      command => 'timedatectl set-local-rtc 0',
      unless  => 'test $(timedatectl show | grep LocalRTC | cut -d= -f2) = no',
      path    => $facts['path'],
    }
  }
```

But that make the hardware clock explicitly managed when it wasn't before.  I don't believe localtime is rational for POSIX system, but I'm not sure just pushing that on folks is the right call.

**Comment by TheMeier** [comment]
I would have preferred `systemd::timedatectl` to be  not funneled through the main class, then the params could be non-optional. But the whole module is designed in that way, and others have different opinions on that ;)

So LGTM...

**Review by TheMeier — APPROVED** [review]

File: manifests/timedatectl.pp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class systemd::timedatectl (
+  Optional[Boolean] $set_local_rtc = $systemd::set_local_rtc,
+  Optional[String[1]] $timezone = $systemd::timezone,
+) {
+  assert_private()
+
+  if $timezone {
+    exec { 'set system timezone':
+      command => "timedatectl set-timezone ${timezone}",
+      unless  => "test $(timedatectl show | grep Timezone | cut -d= -f2) = ${timezone}",
```
Comment (bastelfreak) [inline manifests/timedatectl.pp:27]:
```suggestion
      unless  => "test $(timedatectl show --property Timezone --value) = ${timezone}",
```

Comment (jcpunk) [reply manifests/timedatectl.pp]: Updated in all the snippets.

File: manifests/timedatectl.pp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      path    => $facts['path'],
+    }
+  }
+
+  if $set_local_rtc == true {
```
[the remark this answers is not in the document]
Comment (jcpunk) [reply manifests/timedatectl.pp]: Changed.

File: manifests/timedatectl.pp [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# @api private
+#
+# This class provides an abstract way to set elements with timedatectl
```
Comment (kenyon) [inline manifests/timedatectl.pp:3]:
```suggestion
# @summary This class provides an abstract way to set elements with timedatectl
```

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -386,6 +386,43 @@
...
 
+        context 'when setting timezone' do
+          let(:params) do
+            {
+              timezone: 'America/Chicago',
+            }
+          end
+
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_exec('set system timezone') }
```
Comment (kenyon) [inline spec/classes/init_spec.rb:397]: Update to make use of the param:

```suggestion
          it { is_expected.to contain_exec('set system timezone').with_command('timedatectl set-timezone America/Chicago') }
```

**Comment by jcpunk** [comment]
Changes merged.

**Review by kenyon — APPROVED** [review]
Reference needs a regen, otherwise 👍

**Comment by jcpunk** [comment]
In theory this is sorted.

**Pull request merged**